### PR TITLE
[WIP] add interfaces subtree and switch headers POC to using shared proto files

### PR DESCRIPTION
**Pull request by AlexeyAkhunov**

### CODE REVIEW

File: interfaces/p2psentry/control.proto [hunk 1]
```diff
@@ -21,8 +19,7 @@ message InboundMessage {
 
 message Forks {
   bytes genesis = 1;
-  repeated uint64 passed = 2;
-  uint64 next = 3;
+  repeated uint64 forks = 2;
```
Comment (AskAlexSharov) [inline interfaces/p2psentry/control.proto:22]: ForkID is uint32

File: interfaces/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (AskAlexSharov) [inline interfaces/LICENSE:1]: Does GPL allow to use another license in git subtree?

File: interfaces/p2psentry/control.proto (hunk 1 above)
Comment (vorot93) [reply interfaces/p2psentry/control.proto:22]: It's not ForkId, but fork block numbers. Sentry calculates ForkId and builds fork filter based on this information.

File: interfaces/LICENSE (hunk 2 above)
Comment (AlexeyAkhunov) [reply interfaces/LICENSE:1]: Yes, but perhaps not the other way around

Comment (vorot93) [reply interfaces/LICENSE:1]: The thing with licenses is that the derivative work may be _more_ restrictive than its components.

turbo-geth (GPLv3) is more restrictive than interfaces or rust-sentry (Apache-2.0) so it's OK.

**Pull request merged**